src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -336,6 +338,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     internal var superclass: TypeName = ANY
     internal val superclassConstructorParameters = mutableListOf<CodeBlock>()
     internal val superinterfaces = mutableListOf<TypeName>()
+    internal val superinterfaceDelegates = mutableListOf<ParameterSpec>()
```

Comment:
You can validate that in `build()`, check that for every delegate there indeed is a primary constructor parameter with that name